src/lib.rs

```diff
@@ -935,7 +935,10 @@ impl Drop for Node {
 		self.set_context(None);
 
 		unsafe {
-			// YGNodeFree already orphans all its children
+			let parent = internal::YGNodeGetParent(self.inner_node);
+			if parent != 0 as NodeRef {
+				internal::YGNodeRemoveChild(internal::YGNodeGetParent(self.inner_node), self.inner_node);
```

Comment:
This library is currently using revision `a20bde8444474e7a34352a78073de23c26e08fc5` of Yoga.

So currently `YGNodeFree` behaves as the following:

https://github.com/facebook/yoga/blob/a20bde8444474e7a34352a78073de23c26e08fc5/yoga/Yoga.c#L363-L366

This is an older piece of code before Yoga switched over to the C++ version. From what I can see in the new code, calling `YGNodeFree` properly marks the parent as dirty, so once this project migrates over to using the C++ source, this should work properly. But the current C code we're using _does_ remove the child from the parent.

I would prefer if you add a `// TODO - ...` comment to explain that right now we're explicitly marking the node as dirty and when we switch to the C++ version of Yoga, it won't be necessary. And the instead of calling `YGNodeRemoveChild`, call `YGNodeFree` on the parent if it exists.

Reply:
Ah, I see why `YGNodeMarkDirty` won't work:

```
Only leaf nodes with custom measure functions should manually mark themselves as dirty
```

In that case, I agree with your change, but I would still like a `// TODO` comment on it so we can remember to delete it later when the move to the C++ code happens.